light/txpool.go

```diff
@@ -504,15 +506,17 @@ func (self *TxPool) Content() (map[common.Address]types.Transactions, map[common
 func (self *TxPool) RemoveTransactions(txs types.Transactions) {
 	self.mu.Lock()
 	defer self.mu.Unlock()
+
 	var hashes []common.Hash
+	batch := self.chainDb.NewBatch()
 	for _, tx := range txs {
-		//self.RemoveTx(tx.Hash())
 		hash := tx.Hash()
 		delete(self.pending, hash)
-		self.chainDb.Delete(hash[:])
+		batch.Delete(hash.Bytes())
 		hashes = append(hashes, hash)
 	}
 	self.relay.Discard(hashes)
+	batch.Write()
```

Comment:
I'm not really sure what `self.relay.Discard(hashes)`, but I think it's better to write the changes first and then call it. That way we retain the behavior that the writes conclude before calling this method.